addons/membership/models/partner.py

```diff
@@ -0,0 +1,173 @@
+                                for payment in inv.payment_ids:
+                                    if payment.invoice_ids and any(inv.type == 'out_refund' for inv in payment.invoice_ids):
+                                        state = 2
+                                break
+                            elif mstate == 'open' and state != 0:
+                                state = 1
+                            elif mstate == 'cancel' and state != 0 and state != 1:
+                                state = 2
+                            elif (mstate == 'draft' or mstate == 'proforma') and state != 0 and state != 1:
+                                state = 3
+                if state == 4:
+                    for mline in partner_data.member_lines_ids:
+                        if fields.Date.from_string(mline.date_from) < today_date and fields.Date.from_string(mline.date_to) < today_date and mline.date_from <= mline.date_to and (mline.account_invoice_line_id and mline.account_invoice_line_id.invoice_id.state) == 'paid':
+                            state = 5
+                        else:
+                            state = 6
+                if state == 0:
+                    partner_data.membership_state = 'paid'
+                elif state == 1:
+                    partner_data.membership_state = 'invoiced'
+                elif state == 2:
+                    partner_data.membership_state = 'canceled'
+                elif state == 3:
+                    partner_data.membership_state = 'waiting'
+                elif state == 5:
+                    partner_data.membership_state = 'old'
+                elif state == 6:
+                    partner_data.membership_state = 'none'
+            if partner_data.free_member and state != 0:
+                partner_data.membership_state = 'free'
+            if partner_data.associate_member_id:
+                partner_data.membership_state = partner_data.associate_member_id._membership_state()
+            return partner_data.membership_state
+
+    @api.multi
+    def create_membership_invoice(self, product_id=None, datas=None):
+        """ Create Customer Invoice of Membership for partners.
+        @param datas: datas has dictionary value which consist Id of Membership product and Cost Amount of Membership.
+                      datas = {'membership_product_id': None, 'amount': None}
+        """
+        AccountInv = self.env['account.invoice']
+        InvoiceLine = self.env['account.invoice.line']
+        product_id = product_id or datas.get('membership_product_id', False)
+        amount = datas.get('amount', 0.0)
+        invoice_list = []
+        for partner in self:
+            account_id = partner.property_account_receivable_id and partner.property_account_receivable_id.id or False
+            fpos_id = partner.property_account_position_id and partner.property_account_position_id.id or False
+            addr = partner.address_get(['invoice'])
```

Comment:
``account_id = partner.property_account_receivable_id.id`` and ``fpos_id = partner.property_account_position_id.id`` will do the job.